### PR TITLE
TSG: Application Data Blocked

### PR DESCRIPTION
**Pull request by nibanks**

### CODE REVIEW

**Review by anrossi — APPROVED** [review]

File: docs/TSG.md [hunk 1]
```diff
@@ -73,7 +73,32 @@ The error code indicated in this event is completely application defined (type o
 
 ## Why isn't application data flowing?
 
-> TODO
+Application data is exchanged via [Streams](./Streams.md) and queued by the app via [StreamSend](./api/StreamSend.md). The act of queuing data doesn't mean it will be immediately sent to the peer. There are a number of things that can block or delay the exchange. `QUIC_FLOW_BLOCK_REASON` in [quic_trace.h](../src/inc/quic_trace.h) contains the full list of reasons that data may be blocked. Below is a short explaination of each:
```
Comment (nigriMSFT) [inline docs/TSG.md:76]: "explaination" typo

**Comment by nigriMSFT** [comment]
LGTM

File: docs/TSG.md [hunk 2]
```diff
@@ -73,7 +73,32 @@ The error code indicated in this event is completely application defined (type o
...
+
+```
+[0]0004.0F54::2021/05/14-10:30:22.541024000 [Microsoft-Quic][strm][0xC16BA610] Send Blocked Flags: 128
+```
+
+This event indicates that stream `C16BA610` only has the `QUIC_FLOW_BLOCKED_APP` flag, so it is currently blocked because there is no more application data queued to be sent.
+
+![](images/tx-blocked-state.png)
+
+The [QUIC WPA plugin](../src/plugins/wpa/README.md) also supports visualizing these blocked states via the `QUIC TX Blocked State` graph. It allows you to see what flags are blocking the connection as a whole (shown as stream 0) and what is blocking each individual stream, over the lifetime of the whole connection.
```
Comment (rmiller14) [inline docs/TSG.md:99]: This link looks dead.  I'm guessing the destination will be populated later?

**Review by rmiller14 — APPROVED** [review]

File: docs/TSG.md (hunk 2 above)
Comment (nibanks) [reply docs/TSG.md]: Just fixed the link. It was broken in a recent refactor and I had just copied it from somewhere else in the doc not realizing it was broken. Thanks for checking!!

**Pull request merged**